### PR TITLE
adding TYPE_ENUM to Phalcon\Db\Column

### PR DESCRIPTION
**Pull request by mattvb91**
PR to devtools will follow

### CODE REVIEW

**Comment by sergeyklay** [comment]
Hi @mattvb91 
What about other databases?
If someone moves from MySQL to Postgresql what would happen with those enum types.

In my opinion this PR this PR not completed.

We have to support that type virtually for the other database engines supported.

**Comment by mattvb91** [comment]
@sergeyklay do you mean just defaulting it to something like varchar for all the other adapters?

**Comment by sergeyklay** [comment]
As I said above we have to support that type for the other database. For example we can transparent convert from Myslq's enum to Postgres' varchar. I'm not saying what you need to do exactly. I only express my opinion. Perhaps you'll propose a better idea

**Comment by mattvb91** [comment]
@sergeyklay all the adapters default to either varchar or text if a column type isn't supported which in this case would still be ok as the result would still allow insertion of the data into the column.

**Comment by sergeyklay** [comment]
And back?

**Comment by mattvb91** [comment]
Im not sure if im missing something in your question but if porting data back INTO mysql from a data source not supporting enum it would also default the column to varchar.

**Comment by Dublerq** [comment]
Please consider merging this pull request to 2.x. We're waiting for enums since 2014

File: phalcon/db/adapter/pdo/mysql.zep [hunk 1]
```diff
@@ -280,7 +280,10 @@ class Mysql extends PdoAdapter implements AdapterInterface
 			 */
 			if memstr(columnType, "(") {
 				let matches = null;
-				if preg_match(sizePattern, columnType, matches) {
+
+				if definition["type"] == Column::TYPE_ENUM {
+					let definition["size"] = substr(columnType, 5, -1);
```
Comment (mattvb91) [inline phalcon/db/adapter/pdo/mysql.zep]: @sergeyklay this is where we were trying to use a regex but its easier to simply get the contents out of enum(CONTENTS) by doing a substr select. 

So essentially if you have ENUM('val1', 'val2', 'val3') then definition["size"] = "'val1', 'val2', 'val3'"

Does that make more sense?

**Comment by sergeyklay** [comment]
#10216 #11139

**Comment by sergeyklay** [comment]
@mattvb91

**Comment by dcosta-ptc** [comment]
Any news regarding this topic?

**Comment by sergeyklay** [comment]
@dcosta-ptc Will be part of 3.1.x

**Comment by Dublerq** [comment]
I need to find some time to configure zephir and learn how to test/contribute. Looks like I'll need to speed up needed features development myself because waiting more than half a year to get a simple feature or fix is a bit too long.

**Comment by sergeyklay** [comment]
@Dublerq souds fine!

**Comment by mattvb91** [comment]
@sergeyklay @dcosta-ptc @Dublerq sorry guys im not sure why im only now receiving notifications about this... I've rebased the branch and repushed. Looks like your builds are failing? 

Surprised this hadn't been merged 2 years ago as its a pretty minor change from code side.

**Comment by sergeyklay** [comment]
@mattvb91 I'll take a look a bit later. Thank you for contributing.

**Comment by jesuGMZ** [comment]
3.2.1 and problems too. Any plan when it could be?

**Comment by aat2703** [comment]
4.0.0 according to Sergey... Not usable without ENUM type for me... sadly... Migrations are pretty rough atm...

**Comment by ruudboon** [comment]
Hoping that this and some other types like TIME will get in at 4.0

**Comment by niden** [comment]
@mattvb91 Could you be so kind as to resolve the conflict with the changelog and have a look at the tests? It appears there is a syntax error somewhere. After that I can merge this.

I will need to add more tests on this one

**Comment by niden** [comment]
Thank you!

**Pull request merged**